pkg/build/builder/sti.go

```diff
@@ -251,6 +251,10 @@ func (s *S2IBuilder) Build() error {
 				log.V(5).Infof("Incremental image pull failure: %v", err)
 				config.Incremental = false
 				config.IncrementalFromTag = ""
+			} else {
+				// turn off force pull so that we do not attempt to re-pull with the pull creds in the builder.Build
+				// call below
+				s.build.Spec.Strategy.SourceStrategy.ForcePull = false
```

Comment:
among other things, wouldn't this still break for a "normal" s2i build w/ forcePull set?

Reply:
When I look at line 240 here @bparees and the `if config.Incremental` check it makes me thing that this would not apply to a "normal" s2i build with forcePull set.

Or am I misunderstanding what you mean by "normal" s2i build with forcePull set.

Reply:
my concern was that the failure would still occur for a non-incremental build.  but i think i now understand that the failure is caused by buildah attempting to pull the prev-output image (since it's listed in the Dockerfile), using the normal pull creds, and thus failing, so the failure mode would not be applicable to a non-incremental build.

Reply:
comment updated per conversation with @bparees on slack